infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/select/projection/impl/ShorthandProjection.java

```diff
@@ -39,10 +39,17 @@ public final class ShorthandProjection implements Projection {
     private final String owner;
     
     private final Map<String, ColumnProjection> actualColumns = new LinkedHashMap<>();
-    
-    public ShorthandProjection(final String owner, final Collection<ColumnProjection> columnProjections) {
+
+    private final Map<String, Projection> resultSetColumns = new LinkedHashMap<>();
```

Comment:
Can you merge actualColumns and resultSetColumns to one field?

Reply:
actualColumns is used in  'shardingsphere-encrypt-core' project.  

I will confirm whether SHARDINGSPHERE-ENCRYPT-CORE can accept ExpressionProjection&ColumnProjection  instead of ColumnProjection .

then I will reply to you

Reply:
> Can you merge actualColumns and resultSetColumns to one field?
In some cases, there just only need to use ColumnProjection. It may be better to keep a reference  to actualColumns.

Reply:
> Can you merge actualColumns and resultSetColumns to one field?

It‘s done.